### PR TITLE
Add ubuntu 2404 images to CI config

### PR DESCRIPTION
**Pull request by nikola-matic**
The idea here is to introduce the 2404 images to circleCI config (without actually making use of them). @rodiazet is then to make use of these new images in https://github.com/ethereum/solidity/pull/15320, and only for the 3 or so failing jobs (soltest, force release, etc.), which will allow the evmone bump to pass all of the tests and be merged. We can then update the rest of the jobs to make use of 2404 images wherever appropriate, but in another PR.

### CODE REVIEW

File: .circleci/config.yml
Comment (r0qs) [inline .circleci/config.yml:25]: Just to let it documented, this image comes from: https://github.com/ethereum/solidity/pull/15321#issuecomment-2291244090

File: .circleci/config.yml
Comment (r0qs) [inline .circleci/config.yml:29]: And this image comes from here: https://github.com/ethereum/solidity/pull/15370#issuecomment-2324605800

**Review by r0qs — APPROVED** [review]

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml:661]: We generally add and/remove variants other than the default one as needed. So no need to add them already if nothing's using them.

Having them does not hurt, but the config is already quite long :)

Comment (cameel) [reply .circleci/config.yml:661]: Oh, I guess the assumption is that we'll switch all the 2204 ones to them anyway so they'll all be in use soon?

**Review by cameel — APPROVED** [review]

File: .circleci/config.yml
Comment (cameel) [reply .circleci/config.yml:29]: By the way, I think it's a good convention to hide the comments with older hashes we ended up not using by marking them as "outdated".

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml:21]: Why was #15321 even merged without having a matching PR updating CI that passes all checks? Because of this we're in a really weird state now where the dockerfile source does not match what we're running and also it blocks us from doing any other changes to the images that we might need in the meantime (at least without reverting them). Good thing that we apparently did not need any so far.

Unless these are going to be bumped or removed **very** soon, we should really revert these images (with a version bump to avoid reusing the number).

File: .circleci/config.yml
Comment (nikola-matic) [reply .circleci/config.yml:661]: Correct - the first step is to merge this PR. Then @rodiazet will rebase his evmone bump PR and switch a few of the failing soltest jobs to use the 2404 images added here (required for the evmone bump to pass). Once his PR is merged, we can then switch the rest of the 2204 jobs to 2404, and perform any required clean-up (i.e. delete the 2204 images).

File: .circleci/config.yml
Comment (nikola-matic) [reply .circleci/config.yml:21]: This you mean? https://github.com/ethereum/solidity/pull/15320

We had a ton of problems with the new evmone bump (and still do for ossfuzz) while you were off.

Comment (cameel) [reply .circleci/config.yml:21]: Yes, I mean why did we decide to merge #15321 even though #15320 wasn't passing CI? I remember that there were problems and that's fine, just saying that in that case the right procedure is to hold off on merging both.

**Pull request merged**
